### PR TITLE
Add a new sample using new ConstraintLayout

### PR DESCRIPTION
**Pull request by robjperez**

### CODE REVIEW

**Comment by marinaserranomontes** [comment]
LGTM

File: OpenTokSamples/7.Multiparty-ConstraintLayout/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        }
+    }
+}
+
+dependencies {
+    compile 'com.android.support:appcompat-v7:25.3.1'
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
+    compile 'pub.devrel:easypermissions:0.3.1'
+    compile 'com.opentok.android:opentok-android-sdk:2.10.+'
+}
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/build.gradle]: I see you are bumping up versions here and we are leaving the ones we have in the other samples. IMHO I'd would be great to file a ticket to take care of bumping up the version in the other samples.

File: OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/ConstraintSetHelper.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package com.opentok.android.samples.multiparty_constraintlayout;
+
+import android.support.constraint.ConstraintLayout;
+import android.support.constraint.ConstraintSet;
+import android.transition.TransitionManager;
+
+/**
+ * Created by rpc on 11/04/2017.
+ */
+
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/ConstraintSetHelper.java]: nit: Could you get rid of the header added by AS. We do not have them in any other sample so for the sake of having things the same way IMHO we should remote it.

File: OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+    @AfterPermissionGranted(RC_VIDEO_APP_PERM)
+    private void requestPermissions() {
+        String[] perms = {
+                Manifest.permission.INTERNET,
+                Manifest.permission.CAMERA,
+                Manifest.permission.RECORD_AUDIO
+        };
+        if (EasyPermissions.hasPermissions(this, perms)) {
+            startPublisherPreview();
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java]: nit: If you are not calling/using the `startPublisherPreview()` method any other place there is no need for having a method. IMHO The content of `startPublisherPreview()` is small enough to leave so readability is ok is you do not have the method. Anyway If you do not want to get rid of `startPublisherPreview()` is okay too.

File: OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+                public boolean useTextureViews() {
+                    return true;
+                }
+            }).build();
+            mSession.setSessionListener(this);
+            mSession.connect(OpenTokConfig.TOKEN);
+        } else {
+            EasyPermissions.requestPermissions(this, getString(R.string.rationale_video_app), RC_VIDEO_APP_PERM, perms);
+        }
+    }
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java]: nit: New line here please.

File: OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+    private int getResIdForSubscriberIndex(int index) {
+        TypedArray arr = getResources().obtainTypedArray(R.array.subscriber_view_ids);
+        int subId = arr.getResourceId(index, 0);
+        arr.recycle();
+        return subId;
+    }
+
+
+
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java]: nit: Too many blank lines here.

File: OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+        if (mSubscribers.size() > 0) {
+            for (Subscriber subscriber : mSubscribers) {
+                if (subscriber != null) {
+                    mSession.unsubscribe(subscriber);
+                    subscriber.destroy();
+                }
+            }
+        }
+
+        if (mPublisher != null) {
```
Comment (jaoo) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/src/main/java/com/opentok/android/samples/multiparty_constraintlayout/MainActivity.java]: Should not you remove the publisher's view when disconnecting?

File: OpenTokSamples/7.Multiparty-ConstraintLayout/build.gradle (hunk 1 above)
Comment (robjperez) [inline OpenTokSamples/7.Multiparty-ConstraintLayout/build.gradle]: Good idea, I will file a ticket to do so. Let's do it in a different PR

**Pull request merged**